engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -48,4 +50,66 @@ public DbDistributionState(
             distributionKey,
             commandValueAndValueTypeWrapper);
   }
+
+  @Override
+  public void addCommandDistribution(
+      final long distributionKey, final CommandDistributionRecord commandDistributionRecord) {
+    this.distributionKey.wrapLong(distributionKey);
+    commandDistributionRecordColumnFamily.insert(
+        this.distributionKey,
+        new CommandValueAndValueTypeWrapper().wrap(commandDistributionRecord));
+  }
+
+  @Override
+  public void removeCommandDistribution(final long distributionKey) {
+    this.distributionKey.wrapLong(distributionKey);
+    commandDistributionRecordColumnFamily.deleteIfExists(this.distributionKey);
+  }
+
+  @Override
+  public void addPendingDistribution(final long distributionKey, final int partition) {
+    this.distributionKey.wrapLong(distributionKey);
+    partitionKey.wrapInt(partition);
+    pendingDistributionColumnFamily.insert(distributionPartitionKey, DbNil.INSTANCE);
+  }
+
+  @Override
+  public void removePendingDistribution(final long distributionKey, final int partition) {
+    this.distributionKey.wrapLong(distributionKey);
+    partitionKey.wrapInt(partition);
+    pendingDistributionColumnFamily.deleteExisting(distributionPartitionKey);
+  }
+
+  @Override
+  public boolean hasPendingDistribution(final long distributionKey) {
+    this.distributionKey.wrapLong(distributionKey);
+
+    final var hasPending = new MutableBoolean();
+    pendingDistributionColumnFamily.whileEqualPrefix(
+        this.distributionKey,
+        (compositeKey, dbNil) -> {
+          hasPending.set(true);
+          return false;
+        });
+
+    return hasPending.get();
+  }
+
+  @Override
+  public CommandDistributionRecord getCommandDistributionRecord(
+      final long distributionKey, final int partition) {
+    this.distributionKey.wrapLong(distributionKey);
+    partitionKey.wrapInt(partition);
```

Comment:
>`partitionKey.wrapInt(partition);`

❌ This line is unnecessary because the `commandDistributionRecordColumnFamily` doesn't use the `partitionKey`